cmd/pulumi-test-language/providers/namespaced_provider.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package providers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/blang/semver"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+type NamespacedProvider struct {
+	plugin.UnimplementedProvider
+}
+
+var _ plugin.Provider = (*NamespacedProvider)(nil)
+
+func (p *NamespacedProvider) Close() error {
+	return nil
+}
+
+func (p *NamespacedProvider) Configure(
+	context.Context, plugin.ConfigureRequest,
+) (plugin.ConfigureResponse, error) {
+	return plugin.ConfigureResponse{}, nil
+}
+
+func (p *NamespacedProvider) Pkg() tokens.Package {
+	return "namespaced"
+}
+
+func (p *NamespacedProvider) GetSchema(
+	context.Context, plugin.GetSchemaRequest,
+) (plugin.GetSchemaResponse, error) {
+	resourceProperties := map[string]schema.PropertySpec{
+		"value": {
+			TypeSpec: schema.TypeSpec{
+				Type: "boolean",
+			},
+		},
+	}
+	resourceRequired := []string{"value"}
+
+	pkg := schema.PackageSpec{
+		Name:      "namespaced",
+		Version:   "16.0.0",
+		Namespace: "a-namespace",
```

Comment:
I'm not very familiar with the way we setup these tests... It feels painful to have to write all this boilerplate just for the single meaningful like of `Namespace`. Not sure if anything can be done about that.

Reply:
I don't think so.  We could add the namespace to another provider, but that would break existing tests, so we can't split up the change into separate PRs anymore.  While not ideal, it's just some boilerplate that's copied around, so I don't think it's too bad.

Reply:
Cookies for anyone who works out a nice way to get rid of the boilerplate though